tests/unit/lms/services/blackboard_api/client_test.py

```diff
@@ -237,19 +237,110 @@ def test_it(
         self,
         svc,
         basic_client,
-        BlackboardListGroupSetGroups,
-        blackboard_list_group_set_groups,
+        BlackboardListGroups,
+        blackboard_list_groups,
     ):
         group_sets = svc.group_set_groups("COURSE_ID", "GROUP_SET_ID")
 
         basic_client.request.assert_called_once_with(
             "GET",
             "/learn/api/public/v2/courses/uuid:COURSE_ID/groups/sets/GROUP_SET_ID/groups",
         )
-        BlackboardListGroupSetGroups.assert_called_once_with(
-            basic_client.request.return_value
+        BlackboardListGroups.assert_called_once_with(basic_client.request.return_value)
+        assert group_sets == blackboard_list_groups.parse.return_value
+
+
+class TestGroupCourseGroups:
+    def test_it(
+        self,
+        svc,
+        basic_client,
+        BlackboardListGroups,
+        blackboard_list_groups,
+    ):
+        groups = svc.course_groups("COURSE_ID", current_student_own_groups_only=False)
+
+        basic_client.request.assert_called_once_with(
+            "GET",
+            "/learn/api/public/v2/courses/uuid:COURSE_ID/groups",
+        )
+        BlackboardListGroups.assert_called_once_with(basic_client.request.return_value)
+        assert groups == blackboard_list_groups.parse.return_value
+
+    def test_it_with_group_set_id(self, svc, blackboard_list_groups, groups):
+        blackboard_list_groups.parse.return_value = groups
+
+        groups = svc.course_groups(
+            "COURSE_ID", "OTHER_GROUP_SET", current_student_own_groups_only=False
+        )
+
+        assert len(groups) == 1
+        assert groups[0]["groupSetId"] == "OTHER_GROUP_SET"
```

Comment:
Could do this in a single assert:

```suggestion
        assert [group["groupSetId"] for group in groups] == ["OTHER_GROUP_SET"]
```